libpod/runtime_volume_linux.go

```diff
@@ -66,25 +75,62 @@ func (r *Runtime) newVolume(ctx context.Context, options ...VolumeCreateOption)
 		}
 	}
 
-	// Create the mountpoint of this volume
-	volPathRoot := filepath.Join(r.config.Engine.VolumePath, volume.config.Name)
-	if err := os.MkdirAll(volPathRoot, 0700); err != nil {
-		return nil, errors.Wrapf(err, "error creating volume directory %q", volPathRoot)
-	}
-	if err := os.Chown(volPathRoot, volume.config.UID, volume.config.GID); err != nil {
-		return nil, errors.Wrapf(err, "error chowning volume directory %q to %d:%d", volPathRoot, volume.config.UID, volume.config.GID)
-	}
-	fullVolPath := filepath.Join(volPathRoot, "_data")
-	if err := os.MkdirAll(fullVolPath, 0755); err != nil {
-		return nil, errors.Wrapf(err, "error creating volume directory %q", fullVolPath)
-	}
-	if err := os.Chown(fullVolPath, volume.config.UID, volume.config.GID); err != nil {
-		return nil, errors.Wrapf(err, "error chowning volume directory %q to %d:%d", fullVolPath, volume.config.UID, volume.config.GID)
-	}
-	if err := LabelVolumePath(fullVolPath); err != nil {
-		return nil, err
+	// Now we get conditional: we either need to make the volume in the
+	// volume plugin, or on disk if not using a plugin.
+	if volume.plugin != nil {
+		// We can't chown, or relabel, or similar the path the volume is
```

Comment:
Might be worth breaking the two branches into separate functions. The branches are big and have quite some comments.